src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -518,31 +520,59 @@ impl HummockEventHandler {
     }
 
     fn resolve_version_update_info(
-        pinned_version: PinnedVersion,
+        mut pinned_version: PinnedVersion,
         version_payload: HummockVersionUpdate,
         mut sst_delta_infos: Option<&mut Vec<SstDeltaInfo>>,
     ) -> Option<PinnedVersion> {
-        let newly_pinned_version = match version_payload {
+        match version_payload {
             HummockVersionUpdate::VersionDeltas(version_deltas) => {
-                let mut version_to_apply = (*pinned_version).clone();
-                for version_delta in &version_deltas {
-                    assert_eq!(version_to_apply.id, version_delta.prev_id);
-                    if let Some(sst_delta_infos) = &mut sst_delta_infos {
-                        sst_delta_infos.extend(
-                            version_to_apply
-                                .build_sst_delta_infos(version_delta)
-                                .into_iter(),
+                let mut version_to_apply = pinned_version.version().clone();
+                let table_change_log_to_apply = pinned_version.take_change_log();
```

Comment:
Instead of `take_change_log`, we can have method `change_log_write_lock` to return the write lock on the `table_change_log`. And then in `new_pin_version_with_table_change_log`, we don't have to pass the `table_change_log_to_apply` again. We can use the original `table_change_log` wrapped by `Arc` and simply clone the `Arc`.

Besides, it seems that the parameter `pinned_version` of the current method is not necessary to take the ownership. We can change to pass `&PinnedVersion`, so that the caller can avoid doing clone on it.